tests/lib/snaps/test-snapd-uhid/uhid-test.c

```diff
@@ -0,0 +1,191 @@
+    0x19, 0x01, /* USAGE_MINIMUM (Button 1) */
+    0x29, 0x03, /* USAGE_MAXIMUM (Button 3) */
+    0x15, 0x00, /* LOGICAL_MINIMUM (0) */
+    0x25, 0x01, /* LOGICAL_MAXIMUM (1) */
+    0x95, 0x03, /* REPORT_COUNT (3) */
+    0x75, 0x01, /* REPORT_SIZE (1) */
+    0x81, 0x02, /* INPUT (Data,Var,Abs) */
+    0x95, 0x01, /* REPORT_COUNT (1) */
+    0x75, 0x05, /* REPORT_SIZE (5) */
+    0x81, 0x01, /* INPUT (Cnst,Var,Abs) */
+    0x05, 0x01, /* USAGE_PAGE (Generic Desktop) */
+    0x09, 0x30, /* USAGE (X) */
+    0x09, 0x31, /* USAGE (Y) */
+    0x09, 0x38, /* USAGE (WHEEL) */
+    0x15, 0x81, /* LOGICAL_MINIMUM (-127) */
+    0x25, 0x7f, /* LOGICAL_MAXIMUM (127) */
+    0x75, 0x08, /* REPORT_SIZE (8) */
+    0x95, 0x03, /* REPORT_COUNT (3) */
+    0x81, 0x06, /* INPUT (Data,Var,Rel) */
+    0xc0, /* END_COLLECTION */
+    0xc0, /* END_COLLECTION */
+    0x05, 0x01, /* USAGE_PAGE (Generic Desktop) */
+    0x09, 0x06, /* USAGE (Keyboard) */
+    0xa1, 0x01, /* COLLECTION (Application) */
+    0x85, 0x02, /* REPORT_ID (2) */
+    0x05, 0x08, /* USAGE_PAGE (Led) */
+    0x19, 0x01, /* USAGE_MINIMUM (1) */
+    0x29, 0x03, /* USAGE_MAXIMUM (3) */
+    0x15, 0x00, /* LOGICAL_MINIMUM (0) */
+    0x25, 0x01, /* LOGICAL_MAXIMUM (1) */
+    0x95, 0x03, /* REPORT_COUNT (3) */
+    0x75, 0x01, /* REPORT_SIZE (1) */
+    0x91, 0x02, /* Output (Data,Var,Abs) */
+    0x95, 0x01, /* REPORT_COUNT (1) */
+    0x75, 0x05, /* REPORT_SIZE (5) */
+    0x91, 0x01, /* Output (Cnst,Var,Abs) */
+    0xc0, /* END_COLLECTION */
+};
+
+static int uhid_write(int fd, const struct uhid_event* ev)
+{
+    ssize_t ret;
+
+    ret = write(fd, ev, sizeof(*ev));
+    if (ret < 0) {
+        fprintf(stderr, "Cannot write to uhid: %m\n");
+        return -1;
+    } else if (ret != sizeof(*ev)) {
+        fprintf(stderr, "Wrong size written to uhid: %ld != %lu\n",
+            ret, sizeof(ev));
+        return -1;
+    } else {
+        return 0;
+    }
+}
+
+static int create(int fd)
+{
+    struct uhid_event ev = {
+        .type = UHID_CREATE,
+        .u = {.create.rd_data = rdesc,
+            .create.rd_size = sizeof(rdesc),
+            .create.bus = BUS_USB,
+            .create.vendor = 0x15d9,
+            .create.product = 0x0a37,
+            .create.version = 0,
+            .create.country = 0 }
+
+    };
+    strcpy((char*)ev.u.create.name, "test-uhid-device");
+
+    return uhid_write(fd, &ev);
+}
+
+static void destroy(int fd)
+{
+    struct uhid_event ev;
+
+    memset(&ev, 0, sizeof(ev));
+    ev.type = UHID_DESTROY;
+
+    uhid_write(fd, &ev);
+}
+
+int main(int argc, char** argv)
+{
+    int fd;
+    const char* path = "/dev/uhid";
+    int ret;
+
+    fprintf(stdout, "Open uhid-cdev %s\n", path);
+    fd = open(path, O_RDWR | O_CLOEXEC);
+    if (fd < 0) {
+        fprintf(stderr, "Cannot open uhid-cdev %s: %m\n", path);
+        return EXIT_FAILURE;
+    }
+
+    fprintf(stdout, "Create uhid device\n");
+    ret = create(fd);
+    if (ret) {
```

Comment:
Small nitpick: Since create returns the return value of uhid_write, which returns 0 on success, can you please make this `ret != 0` for clarity.